twisted/trial/util.py

```diff
@@ -362,7 +362,7 @@ def _unusedTestDirectory(base):
             # Create it anew and mark it as ours so the next _removeSafely on
             # it succeeds.
             testdir.makedirs()
-            testdir.child('_trial_marker').setContent(b'')
```

Comment:
Looking at the error at http://twistedmatrix.com/trac/ticket/8577 I don't think this is the correct fix. It looks like `setContent` is broken with `unicode` paths on python3 (probably due to `temporarySibling` generating a `bytes` path always). Rather than fix this one particular caller, `setContent` itself should be fixed.